cds_migrator_kit/videos/weblecture_migration/load/load.py

```diff
@@ -27,9 +50,162 @@ def _prepare(self, entry):
+        video_bucket = Bucket.get(video_deposit["_buckets"]["deposit"])
+        video_deposit_id = video_deposit["_deposit"]["id"]
+
+        # Upload video (main video)
+        bucket_id = video_deposit["_buckets"]["deposit"]
+        # dummy test video, files are not implemented yet
+        video_path = (
+            "cds_migrator_kit/videos/weblecture_migration/data/videos/test_video.mp4"
+        )
+        object_version = self._upload_video_file_to_bucket(
+            bucket_id=bucket_id, video_path=video_path
+        )
+
+        # Create tags, flow tasks and run
+        flow = self._run_flow(object_version, video_deposit_id)
+
+        # Publish video
+        self._publish_video_record(deposit_id=video_deposit_id)
+
+    def _run_flow(self, object_version, deposit_id):
+        """Low level represantation of post api/flows.
+
+        This methods creates updates ObjectVersionTag,
+        Creates the flow with tasks:
+        - ExtractMetadataTask
+        - ExtractFramesTask
+        and runs these tasks sync
+        """
+        deposit_id = str(deposit_id)
+        payload = dict(
+            version_id=str(object_version.version_id),
+            key=object_version.key,
+            bucket_id=str(object_version.bucket_id),
+            deposit_id=str(deposit_id),
+        )
+        # Create FlowMetadata
+        flow = FlowMetadata.create(
+            deposit_id=deposit_id,
+            user_id=2,  # TODO It's failing vithout user id
+            payload=payload,
+        )
+
+        # create/update the object version with ObjectVersionTag, tags the master
+        object_version = init_object_version(flow, has_remote_file_to_download=None)
+        version_id = str(object_version.version_id)
+        flow.payload["version_id"] = version_id
+        db_flag_modified(flow, "payload")
+        db.session.commit()
+
+        # Create flow with tasks: extract metadata, frames
+        celery_tasks = self._create_flow(flow, payload)
+        # Apply tasks sync
+        result = celery_tasks.apply()
+        # Flow and Tasks modifications need to be persisted
+        db.session.commit()
+        index_deposit_project(deposit_id)
+        return flow
+
+    def _create_flow(self, flow, payload):
+        """
+        Creates the flow with celery tasks.
+
+        Instead of creating tasks, if we call:
+            FlowService(flow).run()
+        This is gonna create all the tasks(extract metadata, frames, subformats),
+        and run them async
+        """
+        payload["flow_id"] = str(flow.id)
+        celery_tasks = self._create_flow_tasks(payload)
+        celery_tasks_signatures = []
+        for celery_task_tuple in celery_tasks:
+            assert isinstance(celery_task_tuple, tuple)
+            celery_task, kwargs = celery_task_tuple
+            signature = AVCFlowCeleryTasks.create_task_signature(celery_task, **kwargs)
+            celery_tasks_signatures.append(signature)
```

Comment:
Minor: I think it could be simplified to something like this (not sure why we need the assert)
```suggestion
        celery_tasks_signatures = [
            AVCFlowCeleryTasks.create_task_signature(task, **kwargs)
            for task, kwargs in tasks
        ]

```

Reply:
I'll simply, thanks! Assert also comes from [here](https://github.com/CERNDocumentServer/cds-videos/blob/805a40d35b5774b531a62afaef6140d4f73afa94/cds/modules/flows/api.py#L137)